src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/CustomJsonTypeInfoOfT.cs

```diff
@@ -31,11 +31,7 @@ internal CustomJsonTypeInfo(JsonSerializerOptions options)
         private static JsonConverter GetConverter(JsonSerializerOptions options)
         {
             DefaultJsonTypeInfoResolver.RootDefaultInstance();
-            return GetEffectiveConverter(
```

Comment:
consider removing GetEffectiveConverter, I think it ended up being equivalent to `options.GetConverterForType(typeof(T))`